app/assets/javascripts/requests/requests.js

```diff
@@ -2,6 +2,11 @@
 // All this logic will automatically be available in application.js.
 $(document).ready(function() {
 
+    $( "#returnToRecordLink" ).click(function( event ) {
```

Comment:
Since we do have the "source" param around and the record ID in question couldn't we use that rather than relying on the browser? The only possible client systems for this are blacklight and the main catalog. Both of which will let you do a link back to a given ID value.